feat: Add Instagram URL rewriter to kkinstagram.com

privacydied opened the pull request:
Implements a new Instagram handler that rewrites instagram.com URLs to kkinstagram.com for improved Discord embeds on Posts and Reels.

Changes:
- Add Instagram.cs handler with URL pattern matching and rewriting
- Support instagram.com, www.instagram.com, m.instagram.com, l.instagram.com hosts
- Handle /p/ (posts), /reel/, and /reels/ paths
- Preserve query parameters and URL fragments during rewrite
- Register Instagram handler in Program.cs DI container
- Update README with Instagram support documentation

Implementation details:
- Zero external API calls (pure URL transformation)
- Idempotent: skips already-rewritten kkinstagram.com URLs
- Security: validates exact host match to prevent spoofing
- Follows existing handler patterns (similar to Reddit.cs)
- Can be disabled via Bot:DisabledSites configuration

Example: https://instagram.com/p/ABC123/ → https://kkinstagram.com/p/ABC123/
	new file:   INSTAGRAM_IMPLEMENTATION_SUMMARY.md
	modified:   README.md
	new file:   SaucyBot.Tests/Unit/Site/InstagramTest.cs
	modified:   SaucyBot/Program.cs
	new file:   SaucyBot/Site/Instagram.cs

privacydied committed:
feat: Add Instagram URL rewriter to kkinstagram.com

Implements a new Instagram handler that rewrites instagram.com URLs
to kkinstagram.com for improved Discord embeds on Posts and Reels.

Changes:
- Add Instagram.cs handler with URL pattern matching and rewriting
- Support instagram.com, www.instagram.com, m.instagram.com, l.instagram.com hosts
- Handle /p/ (posts), /reel/, and /reels/ paths
- Preserve query parameters and URL fragments during rewrite
- Register Instagram handler in Program.cs DI container
- Update README with Instagram support documentation

Implementation details:
- Zero external API calls (pure URL transformation)
- Idempotent: skips already-rewritten kkinstagram.com URLs
- Security: validates exact host match to prevent spoofing
- Follows existing handler patterns (similar to Reddit.cs)
- Can be disabled via Bot:DisabledSites configuration

Example: https://instagram.com/p/ABC123/ → https://kkinstagram.com/p/ABC123/
	new file:   INSTAGRAM_IMPLEMENTATION_SUMMARY.md
	modified:   README.md
	new file:   SaucyBot.Tests/Unit/Site/InstagramTest.cs
	modified:   SaucyBot/Program.cs
	new file:   SaucyBot/Site/Instagram.cs

Sn0wCrack commented on SaucyBot/Site/Instagram.cs:
```diff
@@ -0,0 +1,89 @@
+        {
+            _logger.LogDebug("Instagram URL already rewritten to kkinstagram.com, skipping: {Url}", originalUrl);
+            return null;
+        }
```
The `host` group of the regex match will never contain "kkinstagram.com" as the regex cannot match against it: https://regex101.com/r/tQqzpB/1

Additionally even if it did, this is made redundant by your `IsValidInstagramHost` call below.

Sn0wCrack commented on SaucyBot/Site/Instagram.cs:
```diff
@@ -0,0 +1,89 @@
+        {
+            _logger.LogDebug("Invalid Instagram host detected, skipping: {Host}", host);
+            return null;
+        }
```
Similar to above, the regex will never capture anything other than www.instagram.com, instagram.com, m.instagram.com or l.instagram.com. So I think this check is redundant.

Sn0wCrack commented on SaucyBot/Site/Instagram.cs:
```diff
@@ -0,0 +1,89 @@
+        {
+            _logger.LogDebug("Instagram path not in accepted list (posts/reels), skipping: {Path}", path);
+            return null;
+        }
```
This is also checked in your regex by explicitly matching against p, reel and reels at the start of the path portion of your match.

Sn0wCrack commented on INSTAGRAM_IMPLEMENTATION_SUMMARY.md:
I'm not really sure what the point of this empty markdown file is?

privacydied committed:
refactor(instagram): finalize regex-driven rewrite; remove redundant checks; add comprehensive tests

Per PR #444 review feedback:
- Remove unreachable kkinstagram.com guard (regex never matches it)
- Remove redundant IsValidInstagramHost() method (regex constrains hosts)
- Remove redundant IsAcceptedPath() method (regex restricts paths)
- Remove l.instagram.com support (link shim uses /?u= not /p/ or /reel/)
- Keep single debug log for successful rewrites

Added comprehensive unit tests:
- Positive: posts, reels, query/fragment preservation, trailing slashes
- Negative: already rewritten, non-supported endpoints, look-alike hosts
- Edge cases: case insensitivity, extra path segments, mixed content

Implementation:
- 40 lines total (down from 90)
- Regex-only gating for host and path
- Zero external calls, no changes to other handlers
- Preserves path, query, and fragment exactly
	deleted:    INSTAGRAM_IMPLEMENTATION_SUMMARY.md
	modified:   SaucyBot.Tests/Unit/Site/InstagramTest.cs
	modified:   SaucyBot/Site/Instagram.cs

privacydied committed:
refactor(instagram): make Process non-async; preserve regex-only rewrite to kkinstagram

privacydied commented:
refactor(instagram): finalize regex-driven rewrite; remove redundant checks; add comprehensive tests

Per PR #444 review feedback:
- Remove unreachable kkinstagram.com guard (regex never matches it)
- Remove redundant IsValidInstagramHost() method (regex constrains hosts)
- Remove redundant IsAcceptedPath() method (regex restricts paths)
- Remove l.instagram.com support (link shim uses /?u= not /p/ or /reel/)
- Keep single debug log for successful rewrites

Added comprehensive unit tests:
- Positive: posts, reels, query/fragment preservation, trailing slashes
- Negative: already rewritten, non-supported endpoints, look-alike hosts
- Edge cases: case insensitivity, extra path segments, mixed content

Implementation:
- 40 lines total (down from 90)
- Regex-only gating for host and path
- Zero external calls, no changes to other handlers
- Preserves path, query, and fragment exactly

Sn0wCrack reviewed (approved):
Looks good now. Thanks for the contribution.

I'll give this a test myself before deploying as well.

privacydied commented:
no problem dude enjoy
